### PR TITLE
Fix batchnorm gamma

### PR DESCRIPTION
**Pull request by roywei**
Fix: #136 
When gamma or beta is None in Keras, mxnet backend will fail.
Since mx.sym.BatchNorm does not take gamma or beta as None, we still need to create those symbols using default values (ones for gamma, zeros for beta) and set them as not learnable.
There is also option to set that in MXNet Symbol (mx.sym.Variable), but it's easier to do that in Keras normalization layer and both method would break Keras code.

@sandeep-krishnamurthy @kalyc

### CODE REVIEW

File: keras/layers/normalization.py [hunk 1]
```diff
@@ -111,15 +111,35 @@ def build(self, input_shape):
                                          regularizer=self.gamma_regularizer,
                                          constraint=self.gamma_constraint)
         else:
-            self.gamma = None
+            # mxnet backend does not take gamma symbol as None, initialize as ones
+            # and set it as not trainable
+            if K.backend() == 'mxnet':
+                self.gamma = self.add_weight(shape=shape,
+                                             name='gamma',
+                                             initializer=self.gamma_initializer,
```
Comment (sandeep-krishnamurthy) [inline keras/layers/normalization.py]: is gamma_initializer creating all ones and beta_initializer all zeros?
As we were discussing offline, how are you making sure this parameter is not learnt?
What is the effect on model saving and loading - if users are setting scale/center to None, what is saved in trained model?

**Comment by sandeep-krishnamurthy** [comment]
Thanks @roywei for enabling tests.
What is the behavior of saving and loading the model with BatchNorm layer with no center, no scale?

**Comment by roywei** [comment]
@sandeep-krishnamurthy Modified `example/cifar10_resnet_multi_gpu.py` to use `BatchNorm(scale=False, center=False)`.
Trained for 20 epochs,  and loading an saved model for evaluation, got the following result which is the same loss and acc from training output. Verified same behavior as using TensorFlow backend.

highest acc during 20 epochs:
```
Epoch 18/20
Learning rate:  0.001
50000/50000 [==============================] - 27s 535us/step - loss: 0.4444 - acc: 0.8800 - val_loss: 0.8813 - val_acc: 0.7551

Epoch 00018: val_acc improved from 0.73530 to 0.75510, saving model to /home/ubuntu/keras/examples/saved_models/cifar10_ResNet56v1_model.018.h5

```

loading the model back and verify acc
```
model=keras.models.load_model('cifar10_ResNet56v1_model.018.h5')
scores = model.evaluate(x_test, y_test, verbose=1)
print('Test loss:', scores[0])
print('Test accuracy:', scores[1])
```

```
Test loss: 0.8804489216327668
Test accuracy: 0.7549
```

**Comment by roywei** [comment]
Updated CI to actually checkout the Pull Request code instead of running test on master branch. CI passed

**Comment by sandeep-krishnamurthy** [comment]
@roywei - Please add this change in our issue to track changes made in core keras layers.
